pkg/cli/start.go

```diff
@@ -24,58 +32,142 @@ var startCmd = &cobra.Command{
 	Start the telemetry server, which will start listening for data on
 	the input device specified via the --source flags
 	`,
-	Example: `  telemetry start --schema=/foo.asciipb [--port=port] [--fake] [--tty=/dev/...] [--db=a.db]`,
-	RunE:    runStart,
+	Example: `
+	telemetry start --schema=/foo.asciipb
+									[--port=port]
+									[--source=(s|f|r)]
+									[--rest]
+									[--tty=/dev/...]
+									[--db=a.db]
+	`,
+	RunE: runStart,
 }
 
 // ErrorCode is the value to be used by main() as exit code in case of
 // error. For most errors 1 is appropriate, but a signal termination
 // can change this.
 var ErrorCode = 1
 
-var fake bool
-var serverPort int
-var ttyPort string
-var dbName string
-var schemaFile string
+var envPrefix = "uwmidsuntelem"
 
 func init() {
-	startCmd.Flags().StringVarP(&schemaFile, "schema", "s", "", "s")
-	startCmd.Flags().IntVarP(&serverPort, "port", "p", 8080, "port")
-	startCmd.Flags().BoolVarP(&fake, "fake", "f", false, "fake")
-	startCmd.Flags().StringVarP(&ttyPort, "tty", "t", "/dev/ttyUSB0", "tty")
-	startCmd.Flags().StringVarP(&dbName, "db", "d", "", "db")
+	startCmd.Flags().String("schema", "", "The location of the CAN message schema")
+	startCmd.Flags().Int("serverPort", 8080, "The port to setup the HTTP port on")
+	startCmd.Flags().String("interface", "/dev/ttyUSB0", "The interface to listen on")
+	startCmd.Flags().String("dbConnectString", "", "The connect string to use for the database")
+	startCmd.Flags().String("token", "", "Permanently add a new token to the sqlite auth table")
+	startCmd.Flags().String("source", "s", "Source of CAN messages. Can be a combination of (s)erial, (r)est, or (f)ake")
+	startCmd.Flags().String("dbDriver", "sqlite3", "The DB driver to use")
+	startCmd.Flags().String("remoteurl", "", "If CAN data should be sent to a remote server")
+
+	viper.SetEnvPrefix(envPrefix)
+	viper.AutomaticEnv()
+	viper.BindPFlags(startCmd.Flags())
 }
 
-func setupURLRouting(r *chi.Mux, messageBus *pubsub.MessageBus) {
-	r.Get("/ws", ws.ServeHTTP(messageBus, ttyPort, fake))
+func setupFileServer(r *chi.Mux, endpoint string, filesDir string) {
+	if !strings.HasSuffix(endpoint, "/") {
+			panic("Endpoint must be a folder type")
+	}
+	fileServer := http.StripPrefix(
+		endpoint,
+	  http.TimeoutHandler(http.FileServer(http.Dir(filesDir)), 5 * time.Second, "Failed to serve file"),
+	)
+	log.Infof("Mounting fileserver from %s", filesDir)
+	endpoint += "*"
```

Comment:
It's a bit hard to read this way. Could this be done in a separate method to make it easier to read (e.g.;  normalize_endpoint).

Reply:
I've considered refactoring this function but reached the conclusion that it is ok as is. This is because the processing that needs to take place in order to make this URL valid for a fileserver would not be reused in other places.

Besides, it is already in a function and it is performing tasks necessary for that function to succeed.

However I have added some whilespace to make it a bit more readable

Reply:
In terms of readability, I mean that it isn't clear why the star was being added (I figure its for pattern matching but it took me time to look into what it was doing). This is a nit comment so it's not a big deal but just for future reference.